src/Microsoft.AspNet.Mvc.Core/Description/DefaultApiDescriptionProvider.cs

```diff
@@ -537,29 +565,30 @@ private bool Visit(ModelMetadata modelMetadata, BindingSource ambientSource, str
                 //  Order - source: Body
                 //
 
-                var unboundProperties = new HashSet<ModelMetadata>();
+                var unboundProperties = new HashSet<ApiDescriptorBindingContext>();
 
                 // We don't want to append the **parameter** name when building a model name.
                 var newContainerName = containerName;
                 if (modelMetadata.ContainerType != null)
                 {
-                    newContainerName = GetName(containerName, modelMetadata);
+                    newContainerName = GetName(containerName, bindingContext);
                 }
 
                 foreach (var propertyMetadata in modelMetadata.Properties)
                 {
                     var key = new PropertyKey(propertyMetadata, source);
 
+                    var newApiDescriptorContext = ApiDescriptorBindingContext.GetContext(propertyMetadata, null, null);
```

Comment:
Call this something like `propertyContext` the way the naming is done here is less clear with `new`